src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs

```diff
@@ -686,15 +682,7 @@ private void UpdateTextToValue()
 				{
 					// Rounding the value here will prevent displaying digits caused by floating point imprecision.
 					var roundedValue = m_displayRounder.RoundDouble(value);
-
-					if (ApiInformation.IsTypePresent(NumberFormatter.GetType().FullName))
-					{
-						newText = NumberFormatter.FormatDouble(roundedValue);
-					}
-					else
-					{
-						newText = roundedValue.ToString($"0." + new string('#', 6), CultureInfo.CurrentCulture);
-					}
+					newText = NumberFormatter.FormatDouble(roundedValue);
```

Comment:
Interesting! Can you add a runtime test for this section?

Reply:
@morning4coffe-dev the runtime test would init a `NumberBox` control, set its formatter to some funky custom implementation of `INumberFormatter` and then check if it does apply that custom formatting